docs/designs/node-resource-fit-plus-scoring.md

```diff
@@ -0,0 +1,127 @@
+# Enhanced NodeResourceFit Plugin
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+- [Design Consideration](#design-consideration)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+- [Design Details](#design-details)
+  - [NodeResourcesFitPlus](#noderesourcesfitplus)
+  - [ScarceResourceAvoidance](#scarceresourceavoidance)
+  - [Test Plan](#test-plan)
+  - [Graduation Criteria](#graduation-criteria)
+    - [Alpha](#alpha)
+    - [Beta](#beta)
+- [Additional explanation](#additional-explanation)
+- [Implementation History](#implementation-history)
+<!-- /toc -->
+
+
+## Summary
+
+The NodeResourcesFit plug-in of native k8s can only adopt a type of strategy for all resources, such as MostRequestedPriority and LeastRequestedPriority. However, in industrial practice, this design does not apply to some scenarios. For example: In AI scenarios, businesses that apply for GPUs prefer to occupy the entire GPU machine first to prevent GPU fragmentation; businesses that apply for CPU & MEM are prioritized and dispersed to non-GPU machines to prevent excessive consumption of CPU & MEM on GPU machines, resulting in real tasks of applying for GPUs. Pending due to insufficient non-GPU resources
+. Therefore, two plugins are extended to solve this common problem.
+
+## Motivation
+case: 
+- GPU tasks take priority over the entire GPU
+- CPU&MEM tasks are distributed to the CPU machine first
+
+## Design Consideration
+
+- The solution is more versatile, not limited to AI clusters or CPU clusters, and not limited to common CPU resources or extended GPU resources.
+
+- Different resource policies can be configured for different cluster types and prioritized in the form of weights.
+
+- Easy to expand
+
+### Goals
+
+- Different types of resources can be configured with different strategies to prioritize them in the form of weights
+
+- Prevent pods that have not applied for scarce resources from being scheduled to nodes with scarce resources.
+
+### Non-Goals
+
+- None.
+
+## Proposal
+
+Extend two plug-ins to meet the above needs
+
+- NodeResourcesFitPlus
+- ScarceResourceAvoidance
+
```

Comment:
Please add some use cases below. They would be essential to justify a new scheduling capability.

FYI, the [Proposal Template](https://github.com/koordinator-sh/koordinator/blob/main/docs/proposals/YYYYMMDD-template.md).

Reply:
I have written the use cases in user-manuals

Reply:
The *Use Cases* means the user story. I suppose in the user manual it is the examples.

Reply:
done